Kernel/Net/IP/IPv6.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024, kleines Filmröllchen <filmroellchen@serenityos.org>
+ * Copyright (c) 2024, sdomi <ja@sdomi.pl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Assertions.h>
+#include <AK/Endian.h>
+#include <AK/IPv6Address.h>
+#include <AK/Types.h>
+#include <Kernel/Net/IP/IP.h>
+
+namespace Kernel {
+
+// Header extensions and special values only.
+// For transport protocol numbers, see Kernel::TransportProtocol
+enum class IPv6NextHeader : u8 {
+    HopByHopOption = 0,
+    Routing = 43,
+    Fragment = 44,
+    NoNextHeader = 59,
+    DestinationOptions = 60,
+};
+
+// https://datatracker.ietf.org/doc/html/rfc8200#section-3
+class [[gnu::packed]] IPv6PacketHeader {
+public:
+    u8 version() const { return m_version; }
+    void set_version(u8 version) { m_version = version; }
+
+    u16 length() const { return m_length; }
+    void set_length(u16 length) { m_length = length; }
+
+    u16 flow_label() const { return m_flow_label; }
+    void set_flow_label(u16 flow_label) { m_flow_label = flow_label; }
+
+    // Aka. TTL
+    u8 hop_limit() const { return m_hop_limit; }
+    void set_hop_limit(u8 hop_limit) { m_hop_limit = hop_limit; }
+
+    u8 next_header() const { return m_next_header; }
+    void set_next_header(u8 next_header) { m_next_header = next_header; }
+
+    IPv6Address const& source() const { return m_source; }
+    void set_source(IPv6Address const& address) { m_source = address; }
+
+    IPv6Address const& destination() const { return m_destination; }
+    void set_destination(IPv6Address const& address) { m_destination = address; }
+
+    void* payload() { return this + 1; }
+    void const* payload() const { return this + 1; }
```

Comment:
I've remade this (and all other relevant structs in `Kernel/Net`) to use `m_payload`; I'm not sure about returning a Span from here, as it would result in a non-insignificant refactor, as those functions are called (and directly cast) in a lot of places.

Redoing everything here to use Spans may be a good idea for a future refactor, but I would prefer to keep it out of this (already quite large) MR.